cognoml/analysis.py

```diff
@@ -1,121 +1,145 @@
+        pipeline = self.pipeline
+        try:
+            pipeline.fit(X=x_train, y=y_train)
+        except AttributeError:
+            print("Pipeline {} does not have a fit method".format(pipeline))
 
-    performance = collections.OrderedDict()
-    for part, df in ('training', obs_train_df), ('testing', obs_test_df):
-        y_true = df.status
-        y_pred = df.predicted_status
-        metrics = utils.class_metrics(y_true, y_pred)
-        metrics.update(utils.threshold_metrics(y_true, y_pred))
-        performance[part] = metrics
-    performance['cv'] = {'auroc': grid_search.best_score_}
-    results['performance'] = performance
-    
-    gs = collections.OrderedDict()
-    gs['cv_scores'] = utils.cv_results_to_df(grid_search.cv_results_)
-    results['grid_search'] = gs
-    
-    results['model'] = utils.model_info(grid_search.best_estimator_.steps[-1][1])
+    def predict(self):
+        """
+        Internal wrapper for scikit-learn's predict method with custom data pipeline
 
-    feature_df = utils.get_feature_df(grid_search, X.columns)
-    results['model']['features'] = feature_df
+        Returns
+        -------
+        predict_df: Pandas data frame
+            Mutation predictions for entire feature dataframe
+        """
+        pipeline = self.pipeline
+        x = self.X
+        try:
+            predict_df = pd.DataFrame({'sample_id': x.index, 'predicted_status': pipeline.predict(x)})
+        except AttributeError:
+            raise AttributeError("Pipeline {} does not have a predict method".format(pipeline))
+        if hasattr(pipeline, 'decision_function'):
+            predict_df['predicted_score'] = pipeline.decision_function(x)
+        if hasattr(pipeline, 'predict_proba'):
+            predict_df['predicted_prob'] = pipeline.predict_proba(x)[:, 1]
+        return predict_df
 
-    results['observations'] = obs_df
-    
-    if json_sanitize:
-        results = utils.make_json_serializable(results)
-    
-    return results
+    def get_results(self):
+        pipeline = self.pipeline
+        results = collections.OrderedDict()
+        x_test = self.x_test
+        x = self.X
+        obs_df = self.obs_df
+        obs_df = pd.DataFrame(collections.OrderedDict((('sample_id', obs_df.index), ('status', obs_df.values))))
+        predict_df = self.predict()
+        obs_df['testing'] = obs_df['sample_id'].isin(x_test.index).astype(int)
+        obs_df = obs_df.merge(predict_df, how='right', sort=True)
+        obs_df['selected'] = obs_df['sample_id'].isin(obs_df['sample_id']).astype(int)
+        for column in 'status', 'testing', 'selected':
+            obs_df[column] = obs_df[column].fillna(-1).astype(int)
+        obs_train_df = obs_df.query("testing == 0")
+        obs_test_df = obs_df.query("testing == 1")
+        dimensions = collections.OrderedDict()
+        dimensions['observations_selected'] = sum(obs_df.selected == 1)
+        dimensions['observations_unselected'] = sum(obs_df.selected == 0)
```

Comment:
The resulting JSON has:

```json
    "observations_unselected": 0,
```

Rather than. 

```json
    "observations_unselected": 2264,
```

See this comment in original code:

```
# obs_df switches to containing non-selected samples
```